source/common/network/resolver_impl.cc

```diff
@@ -64,18 +56,17 @@ InstanceConstSharedPtr resolveProtoAddress(const envoy::api::v2::Address& addres
 
 InstanceConstSharedPtr
 resolveProtoSocketAddress(const envoy::api::v2::SocketAddress& socket_address) {
-  const ResolverFactory* resolver_factory = nullptr;
+  Resolver* resolver = nullptr;
   const std::string& resolver_name = socket_address.resolver_name();
   if (resolver_name.empty()) {
-    resolver_factory = Registry::FactoryRegistry<ResolverFactory>::getFactory(
-        Config::AddressResolverNames::get().IP);
+    resolver =
+        Registry::FactoryRegistry<Resolver>::getFactory(Config::AddressResolverNames::get().IP);
```

Comment:
Is `getFactory` now a misnomer? It's not returning a factory.. I thought it was fine before for the reasons you stated, but it's OK this way, just needs some changes to nomenclature.

Reply:
A Resolver is not a literal factory, but it does produce Address::Instances. Alternatively, we can rename FactoryRegistry and its methods to something more generic like NamedRegistry, since that's the only method classes are required to implement.